wgengine/userspace.go

```diff
@@ -1316,6 +1337,55 @@ func (e *userspaceEngine) mySelfIPMatchingFamily(dst netaddr.IP) (src netaddr.IP
 	return netaddr.IP{}, errors.New("no self address in netmap matching address family")
 }
 
+func (e *userspaceEngine) sendICMPEchoRequest(ip netaddr.IP, peer *tailcfg.Node, res *ipnstate.PingResult, cb func(*ipnstate.PingResult)) {
```

Comment:
change `ip` to `dstIP`? At first read I thought maybe `ip` was the source and `peer` was the destination, but that `peer` really isn't doing much is it?  Fine to keep. I think if the argument is `dstIP` or it had a bit of docs then it wouldn't be confusing.

Reply:
Yeah, I think this might be able to be refactored down, it's tempting to at some point pull the whole set of pings off of endpoint out to a separate object, as well because this is all very large.